git/transport.go

```diff
@@ -0,0 +1,34 @@
+package git
+
+import (
+	"io"
+	"net/url"
+)
+
+const (
+	ServiceReceivePack = "receive-pack"
+	ServiceUploadPack  = "upload-pack"
+)
+
+type GitTransporter interface {
+	GitTransport(vcsType string, cloneURL *url.URL) (GitTransport, error)
```

Comment:
Can we [avoid stutter](https://blog.golang.org/package-names#TOC_3.), since these exported identifiers are in `git` package already, so the users will see:

```Go
git.GitTransporter
git.GitTransport
```

What about:

```Go
git.Transporter
git.Transport
```

Would that cause any naming conflicts?